web-ng/src/app/components/drawing-tools/drawing-tools.component.ts

```diff
@@ -58,13 +59,16 @@ export class DrawingToolsComponent implements OnInit, OnDestroy {
     getPinImageSource(this.green)
   );
   addPointIcon = this.addPointIconBlack;
+  observationId$: Observable<string | null>;
 
   constructor(
     private readonly changeDetectorRef: ChangeDetectorRef,
     private drawingToolsService: DrawingToolsService,
     private sanitizer: DomSanitizer,
+    private navigationService: NavigationService,
     projectService: ProjectService
   ) {
+    this.observationId$ = this.navigationService.getObservationId$();
```

Comment:
It's not immediately clear what view state this is used for in the template and controller; this might be clearer: 
```ts
disableControls: Observable<boolean>;
...
this.disableControls$ = this.navigationService.getObservationId$().pipe(map(obs => !!obs));
```